[Bug]: Tutorial IK_motion_planning_grasp.py fails with RigidSolver._func_forward_kinematics_entity() argument mismatch

### Bug Description

When running the example `examples/tutorials/IK_motion_planning_grasp.py` from the latest main branch,
I get a TaichiTypeError due to a mismatch in `_func_forward_kinematics_entity()` arguments.


### Steps to Reproduce

git clone https://github.com/Genesis-Sim/Genesis.git
cd Genesis
pip install -e .[render]

python examples/tutorials/IK_motion_planning_grasp.py


### Expected Behavior

When running the ` IK_motion_planning_grasp.py ` example, the simulation should execute without errors, and the draw_debug_path function should correctly visualize the planned end-effector path based on the OMPL-generated trajectory. The inverse kinematics and forward kinematics calls should match function signatures and execute normally.

### Screenshots/Videos

<img width="2798" height="2560" alt="Image" src="https://github.com/user-attachments/assets/cb22d098-da49-4002-88e2-4c99737dad6f" />

### Relevant log output

```shell
(env_genesis) kentliukx@kentliukx-Jiaolong-Series-MRID6-23:~/Genesis$ /home/kentliukx/miniconda3/envs/env_genesis/bin/python /home/kentliukx/Genesis/examples/tutorials/IK_motion_planning_grasp.py
[Genesis] [23:48:44] [INFO] ╭─────────────────────────────────────────────────────────────────────────────────────╮
[Genesis] [23:48:44] [INFO] │┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉ Genesis ┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉│
[Genesis] [23:48:44] [INFO] ╰─────────────────────────────────────────────────────────────────────────────────────╯
[Genesis] [23:48:44] [INFO] Consider setting 'performance_mode=True' in production to maximise runtime speed, if significantly increasing compilation time is not a concern.
[Genesis] [23:48:44] [INFO] Running on [NVIDIA GeForce RTX 4070 Laptop GPU] with backend gs.cuda. Device memory: 7.60 GB.
[Genesis] [23:48:44] [INFO] 🚀 Genesis initialized. 🔖 version: 0.3.0, 🌱 seed: None, 📏 precision: '32', 🐛 debug: False, 🎨 theme: 'dark'.
[Genesis] [23:48:44] [INFO] Scene <ab77fa2> created.
[Genesis] [23:48:44] [INFO] Adding <gs.RigidEntity>. idx: 0, uid: <9c5e18d>, morph: <gs.morphs.Plane>, material: <gs.materials.Rigid>.
[Genesis] [23:48:44] [INFO] Adding <gs.RigidEntity>. idx: 1, uid: <f1fbb74>, morph: <gs.morphs.Box>, material: <gs.materials.Rigid>.
[Genesis] [23:48:44] [INFO] Adding <gs.RigidEntity>. idx: 2, uid: <3180d53>, morph: <gs.morphs.MJCF(file='/home/kentliukx/Genesis/genesis/assets/xml/franka_emika_panda/panda.xml')>, material: <gs.materials.Rigid>.
[Genesis] [23:48:45] [WARNING] (MJCF) Approximating tendon by joint actuator for `finger_joint1`
[Genesis] [23:48:45] [WARNING] (MJCF) Actuator control gain and bias parameters cannot be reduced to a unique PD control position gain. Using max between gain and bias for joint `finger_joint1`.
[Genesis] [23:48:45] [WARNING] (MJCF) Approximating tendon by joint actuator for `finger_joint2`
[Genesis] [23:48:45] [WARNING] (MJCF) Actuator control gain and bias parameters cannot be reduced to a unique PD control position gain. Using max between gain and bias for joint `finger_joint2`.
[Genesis] [23:48:45] [INFO] Applying offset to base link's pose with user provided value in morph.
[Genesis] [23:48:45] [INFO] Building scene <ab77fa2>...
[Genesis] [23:48:46] [WARNING] Reference robot position exceeds joint limits.
[Genesis] [23:48:46] [WARNING] Constraint solver time constant should be greater than 2*substep_dt. timeconst is changed from `0.005` to `0.02`). Decrease simulation timestep or increase timeconst to avoid altering the original value.
[Genesis] [23:48:48] [INFO] Compiling simulation kernels...
[Genesis] [23:48:53] [INFO] Building visualizer...
[Genesis] [23:48:54] [INFO] Viewer created. Resolution: 3413×2560, max_FPS: 60.
/home/kentliukx/Genesis/genesis/engine/entities/rigid_entity/rigid_entity.py:1339: SyntaxWarning: Calling non-taichi function "self._solver._func_forward_kinematics_entity". Scope inside the function is not processed by the Taichi AST transformer. The function may not work as expected. Proceed with caution! Maybe you can consider turning it into a @ti.func?
  self._solver._func_forward_kinematics_entity(
Traceback (most recent call last):
  File "/home/kentliukx/Genesis/examples/tutorials/IK_motion_planning_grasp.py", line 67, in <module>
    path_debug = scene.draw_debug_path(path, franka)
  File "/home/kentliukx/Genesis/genesis/utils/misc.py", line 146, in wrapper
    return method(self, *args, **kwargs)
  File "/home/kentliukx/Genesis/genesis/engine/scene.py", line 1096, in draw_debug_path
    pos, quat = entity.forward_kinematics(qposs[indices[i]])
  File "/home/kentliukx/Genesis/genesis/utils/misc.py", line 146, in wrapper
    return method(self, *args, **kwargs)
  File "/home/kentliukx/Genesis/genesis/engine/entities/rigid_entity/rigid_entity.py", line 1307, in forward_kinematics
    self._kernel_forward_kinematics(
  File "/home/kentliukx/miniconda3/envs/env_genesis/lib/python3.10/site-packages/taichi/lang/kernel_impl.py", line 1178, in __call__
    raise type(e)("\n" + str(e)) from None
taichi.lang.exception.TaichiTypeError: 
File "/home/kentliukx/Genesis/genesis/engine/entities/rigid_entity/rigid_entity.py", line 1339, in _kernel_forward_kinematics:
            self._solver._func_forward_kinematics_entity(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._idx_in_solver,
                ^^^^^^^^^^^^^^^^^^^^
                envs_idx[i_b_],
                ^^^^^^^^^^^^^^^
                self._solver.links_state,
                ^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.links_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.joints_state,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.joints_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.dofs_state,
                ^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.dofs_info,
                ^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.entities_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver._rigid_global_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver._static_rigid_sim_config,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            )
            ^
TypeError when calling `_func_forward_kinematics_entity`: RigidSolver._func_forward_kinematics_entity() takes 3 positional arguments but 12 were given.

[Genesis] [23:48:58] [ERROR] TaichiTypeError: 
File "/home/kentliukx/Genesis/genesis/engine/entities/rigid_entity/rigid_entity.py", line 1339, in _kernel_forward_kinematics:
            self._solver._func_forward_kinematics_entity(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._idx_in_solver,
                ^^^^^^^^^^^^^^^^^^^^
                envs_idx[i_b_],
                ^^^^^^^^^^^^^^^
                self._solver.links_state,
                ^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.links_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.joints_state,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.joints_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.dofs_state,
                ^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.dofs_info,
                ^^^^^^^^^^^^^^^^^^^^^^^
                self._solver.entities_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver._rigid_global_info,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                self._solver._static_rigid_sim_config,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            )
            ^
TypeError when calling `_func_forward_kinematics_entity`: RigidSolver._func_forward_kinematics_entity() takes 3 positional arguments but 12 were given.
[Genesis] [23:48:58] [INFO] 💤 Exiting Genesis and caching compiled kernels...
```

### Environment

Ubuntu 24.04.2 LTS
Python 3.10.18
taichi 1.7.4
GPU: NVIDIA GeForce RTX 4070m
Driver Version: 575.64.03      
CUDA Version: 12.9








### Release version or Commit ID

commit hash: 4a2987e329769f596e656f88bbecc6a79dfb1d2e

### Additional Context

It's hard to follow the tutorial with the bug existing TAT

--- a/examples/tutorials/IK_motion_planning_grasp.py
+++ b/examples/tutorials/IK_motion_planning_grasp.py
@@ -64,7 +64,7 @@ path = franka.plan_path(
     num_waypoints=200,  # 2s duration
 )
 # draw the planned path
-# path_debug = scene.draw_debug_path(path, franka)
+path_debug = scene.draw_debug_path(path, franka)
 
 # execute the planned path
 for waypoint in path:
@@ -72,7 +72,7 @@ for waypoint in path:
     scene.step()
 
 # remove the drawn path
-# scene.clear_debug_object(path_debug)
+scene.clear_debug_object(path_debug)
 
 # allow robot to reach the last waypoint
 for i in range(100):

--- a/genesis/engine/entities/rigid_entity/rigid_entity.py
+++ b/genesis/engine/entities/rigid_entity/rigid_entity.py
@@ -1326,7 +1326,7 @@ class RigidEntity(Entity):
             # set new qpos
             self._solver.qpos[qs_idx[i_q_], envs_idx[i_b_]] = qpos[i_b_, i_q_]
             # run FK
-            self._solver.func_forward_kinematics_entity(
+            gs.engine.solvers.rigid.rigid_solver_decomp.func_forward_kinematics_entity(
                 self._idx_in_solver,
                 envs_idx[i_b_],
                 self._solver.links_state,
@@ -1343,16 +1343,16 @@ class RigidEntity(Entity):
         ti.loop_config(serialize=self._solver._para_level < gs.PARA_LEVEL.PARTIAL)
         for i_l_, i_b_ in ti.ndrange(links_idx.shape[0], envs_idx.shape[0]):
             for i in ti.static(range(3)):
-                links_pos[i_b_, i_l_, i] = self._solver.links_state.pos[links_idx[i_l_], envs_idx[i_b_]]
+                links_pos[i_b_, i_l_, i] = self._solver.links_state.pos[links_idx[i_l_], envs_idx[i_b_]][i]
             for i in ti.static(range(4)):
-                links_quat[i_b_, i_l_, i] = self._solver.links_state.quat[links_idx[i_l_], envs_idx[i_b_]]
+                links_quat[i_b_, i_l_, i] = self._solver.links_state.quat[links_idx[i_l_], envs_idx[i_b_]][i]
 
         ti.loop_config(serialize=self._solver._para_level < gs.PARA_LEVEL.ALL)
         for i_q_, i_b_ in ti.ndrange(qs_idx.shape[0], envs_idx.shape[0]):
             # restore original qpos
             self._solver.qpos[qs_idx[i_q_], envs_idx[i_b_]] = self._IK_qpos_orig[qs_idx[i_q_], envs_idx[i_b_]]
             # run FK
-            self._solver.func_forward_kinematics_entity(
+            gs.engine.solvers.rigid.rigid_solver_decomp.func_forward_kinematics_entity(
                 self._idx_in_solver,
                 envs_idx[i_b_],
                 self._solver.links_state,

